BSD-48: Social nav story

mejiaj opened the pull request:
Creates social nav component with three variants. Closes #48.

## Variants

**Default**
This one might not be necessary, but including for now.
<img width="150" alt="image" src="https://github.com/Bixal/bixal-site-drupal/assets/3385219/f5aa0172-bb29-435c-b200-ccbe8312a3b2">

**Accent warm** (for content)
<img width="150" alt="image" src="https://github.com/Bixal/bixal-site-drupal/assets/3385219/fd52cde0-6120-412e-af12-9d0b93f9fd46">


**Inverse** (for footer)
<img width="181" alt="image" src="https://github.com/Bixal/bixal-site-drupal/assets/3385219/e2a813f4-5c39-4698-a8ab-ddaec430aa82">

## Settings

- `variant:` String for component variant.
- `aria_label:` String for unique aria labels. Unique label is required to avoid A11Y issues when there are multiple on the same page.
- `items:` Array of social nav objects.
  -   `href:` Link to social network.
  -   `label:` Social network label used for link title.
  -   `icon_path:` Path to static asset directory. Will fallback to social network string if not defined.

mpbixal commented on stories/components/footer/footer.stories.js:
```diff
@@ -7,9 +7,15 @@ import "../social-nav/social-nav.scss";
 import "../button/button.scss";
 import "../section/section.scss";
 
+import SocialNavContent from "../social-nav/social-nav.content.json";
```
@mejiaj since the footer is used on all the pages, don't we need to include this in all their stories.js too?

mejiaj replied:
Yes, great catch. Fixed in d2e140b.

mpbixal commented on stories/components/social-nav/social-nav.content.json:
```diff
@@ -0,0 +1,23 @@
+{
```
@mejiaj The fallback path does not seem to be working for this and the icon_paths are not set here, I'm seeing:
![image](https://github.com/Bixal/bixal-site-drupal/assets/121190622/41575aea-0124-4ec3-ba36-a8631ebd6229)

mejiaj replied:
@mpbixal 

Probably tried being too clever. The intention here was to _try_ to set a good default in StorybookJS. I see now it's not helpful anywhere else. Removed in cddba22.

Open to suggestions on improving this experience in Drupal. I've removed this and explicitly defined the path in `social-nav.content.json`.

---

_I wasn't able to reproduce with `npm run storybook:local` or `http://storybook.bixalcom.lndo.site/`._

mpbixal commented:
@mejiaj please see my last four commits and ensure I did not break anything. I:
* updated the social icons links to match their link
* removed imports from the footer story because I don't believe they are needed, I believe there are  a couple other stories that might need this same treatment.
* also note that I added a default value when one of the components imports another component and that nested component has a 'with'. If Drupal includes the parent component, and the 'with' variable is not defined, the site will 500.

mpbixal commented:
@mejiaj I did a little more research, and muddled through https://storybook.js.org/tutorials/intro-to-storybook/angular/en/composite-component/ and it looks like you can re-use components, my latest commit imports the story instead of the data and css, in this way, I can grab the arguments from the old story and that story imports the css it needs. Let me know your thoughts.

mpbixal commented:
> @mejiaj sorry last thing (I hope), the footer component has another call to social-nav at the very top ![image](https://private-user-images.githubusercontent.com/121190622/299101642-08bb8cb9-5325-4a01-9b2a-b0803a1f9224.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwNDQzMDEsIm5iZiI6MTcwNjA0NDAwMSwicGF0aCI6Ii8xMjExOTA2MjIvMjk5MTAxNjQyLTA4YmI4Y2I5LTUzMjUtNGEwMS05YjJhLWIwODAzYTFmOTIyNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyM1QyMTA2NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MmQ1MDZlZmI4YTMwNDVjZDYyYmI4YzE4NzRhOWIxNTk5YzM5YzkxYWIzODM4NGZiYjU1ZmYyMzRhODQ1NmY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MB499PliP8TZfOshAbaD-x-1RBkjopDQWCNPOwbGG6c) It doesn't get render the content because it's not passed 'with socialNav' like the other reference and it's hidden by a css block as well: .bix-site-footer__primary .bix-social-nav { @include at-media("tablet-lg") { display: none; } } Can we nix both the css and the reference to the social nav?

False alarm, I get it now, this was for the mobile menu. I added the with socialNav to it as well so it renders.

mejiaj commented:
@mpbixal thanks for those updates! 

We're re-using component stories in pages, but I was running into rendering issues with Footer. Probably just a local cache issue or something. On code updates my footer would disappear with _only_ the story. Doesn't seem to be happening anymore though.

I've made the following changes:

1. Removed duplicate social nav in 9ced15e. Discussed in our meeting yesterday, but no need for this one now.
2. I wasn't aware of `null` error. Thanks for that info. I ended up simply checking if data existed in fa7b1f2. This matches  pattern used in other templates. Will keep this technique in mind for future work though, thanks!
3. Adjusted accreditations spacing. Fixing a visual regression and improving spacing between logos in dd6e784. 

---

More Footer refinements coming in #16.

buddy-harlow reviewed (approved):
Looks good to me James! Thanks for taking care of the issues we discussed yesterday in our sync regarding the footer nav as well.
